### PR TITLE
Multiplayer IMGui Option to Host and Launch Local Client

### PR DESCRIPTION
**Pull request by AMZN-Gene**
- Adding command _connect_onstartup_ which can be passed by command-line when opening GameLauncher.exe which calls connect as soon as the MultiplayerSystemComponent is activated
- Uninitialized multiplayer systems will display an IMGui Multiplayer option to 'Host'
image
- Hosts will have an IMGui Multiplayer menu option to 'Launch Local Client'; this will launch the gamelauncher.exe and connect to the host automatically.
image

Feature request #9497

Signed-off-by: Gene Walters <genewalt@amazon.com>

### CODE REVIEW

**Review by AMZN-puvvadar — APPROVED** [review]

File: Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp [hunk 1]
```diff
@@ -1243,8 +1252,56 @@ namespace Multiplayer
     }
     AZ_CONSOLEFREEFUNC(host, AZ::ConsoleFunctorFlags::DontReplicate, "Opens a multiplayer connection as a host for other clients to connect to");
 
-    void connect([[maybe_unused]] const AZ::ConsoleCommandContainer& arguments)
+    void launch_local_client(const AZ::ConsoleCommandContainer& arguments)
     {
+        // Try finding the game launcher from the executable folder where this server was launched from.
+        AZ::IO::FixedMaxPath gameLauncherPath = AZ::Utils::GetExecutableDirectory();
```
Comment (AMZN-Olex) [inline Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp:1258]: Does this work with an installer?

Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp:1258]: good question, was going to try this with the Linux installer; currently the INSTALL target is unknown and the online deb package failed install. I'll keep looking for a way to test this...
This should only be called on a gamelauncher or serverlauncher, not editor, so I think those usually typically live in the same folder even if using the Installer

Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp:1258]: Works with INSTALL! but this comment reminded me i should turn off the HOST option when in Editor :P Updated!

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugSystemComponent.cpp [hunk 2]
```diff
@@ -114,6 +116,29 @@ namespace Multiplayer
...
             ImGui::Checkbox("Multiplayer Audit Trail", &m_displayNetAuditTrail);
+
+            if (auto multiplayerInterface = AZ::Interface<IMultiplayer>::Get(); multiplayerInterface && !m_applicationType.IsEditor())
+            {
+                if (auto console = AZ::Interface<AZ::IConsole>::Get())
+                {
+                    const MultiplayerAgentType multiplayerAgentType = multiplayerInterface->GetAgentType();
+                    if (multiplayerAgentType == MultiplayerAgentType::Uninitialized)
+                    {
+                        if (ImGui::Button("Host"))
```
Comment (lmbr-pip) [inline Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugSystemComponent.cpp:127]: nit: Move UX strings to static consts so we have some hope of localizing in the future?

File: Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp [hunk 3]
```diff
@@ -92,6 +93,7 @@ namespace Multiplayer
         "The base used for blending between network updates, 0.1 will be quite linear, 0.2 or 0.3 will "
         "slow down quicker and may be better suited to connections with highly variable latency");
     AZ_CVAR(bool, bg_multiplayerDebugDraw, false, nullptr, AZ::ConsoleFunctorFlags::Null, "Enables debug draw for the multiplayer gem");
+    AZ_CVAR(bool, connect_onstartup, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Whether to call connect as soon as the Multiplayer SystemComponent is activated.");
```
Comment (lmbr-pip) [inline Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp:96]: Feels like cl_connect_onstartup? Assume client only?

File: Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp [hunk 4]
```diff
@@ -1243,8 +1252,56 @@ namespace Multiplayer
...
+        
+        // Launch GameLauncher and connect to this server
+        bool launchSuccess = AzFramework::ProcessLauncher::LaunchUnwatchedProcess(processLaunchInfo);
+        if (!launchSuccess)
+        {
+            AZLOG_ERROR("Failed to launch the local client process.");
+            return;
+        }
+    }
+    AZ_CONSOLEFREEFUNC(launch_local_client, AZ::ConsoleFunctorFlags::DontReplicate, "Launches a local client to connects to this host (only applies if currently hosting)");
```
Comment (lmbr-pip) [inline Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp:1294]: Should this be sv_launch_local_client? as it only makes sense on the server?

Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp]: Good thinking! fixed!

File: Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp (hunk 3 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp]: yes! fixed!

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugSystemComponent.cpp (hunk 2 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugSystemComponent.cpp]: Fixed!

**Pull request merged**